### PR TITLE
BAVL-25 small changes to fix issues with creation of appointments in prison api.

### PR DESCRIPTION
**Pull request by matt-crowson-moj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/client/prisonapi/PrisonApiClient.kt [hunk 1]
```diff
@@ -37,10 +36,10 @@ class PrisonApiClient(private val prisonApiWebClient: WebClient) {
     startTime: LocalTime,
     endTime: LocalTime,
     comments: String? = null,
-  ): Event? =
+  ): ScheduledEvent? =
     prisonApiWebClient
       .post()
-      .uri("/bookings/{bookingId}/appointments", bookingId)
+      .uri("/api/bookings/{bookingId}/appointments", bookingId)
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/client/prisonapi/PrisonApiClient.kt:42]: First thing that was missing was the `/api` prefix.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/client/prisonapi/PrisonApiClient.kt [hunk 2]
```diff
@@ -52,7 +51,9 @@ class PrisonApiClient(private val prisonApiWebClient: WebClient) {
         ),
       )
       .retrieve()
-      .bodyToMono(Event::class.java)
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/client/prisonapi/PrisonApiClient.kt:55]: Second thing wrong, using incorrect return type.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/client/prisonapi/PrisonApiClient.kt [hunk 3]
```diff
@@ -52,7 +51,9 @@ class PrisonApiClient(private val prisonApiWebClient: WebClient) {
...
-      .onErrorResume(WebClientResponseException.NotFound::class.java) { Mono.empty() }
+      .bodyToMono(ScheduledEvent::class.java)
       .block()
 }
+
+// Overriding due to deserialisation issues from generated type.
+data class ScheduledEvent(val eventId: Long)
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/client/prisonapi/PrisonApiClient.kt:59]: Although not ideal I am comfortable with this, we really only need the event ID for logging purposes anyway.

**Review by timharrison-moj — APPROVED** [review]
LGTM

**Pull request merged**